validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/performance/SyncCommitteePerformanceTracker.java

```diff
@@ -0,0 +1,204 @@
+      final Map<UInt64, Set<UInt64>> producingValidatorsBySlot) {
+
+    final int numberOfExpectedSignatures =
+        assignedSubcommitteeIndicesByValidatorIndex.values().stream().mapToInt(Set::size).sum()
+            * spec.atEpoch(epoch).getSlotsPerEpoch();
+
+    int producedSignatureCount = 0;
+    final List<SafeFuture<Integer>> includedSignatureCountFutures = new ArrayList<>();
+    for (Map.Entry<UInt64, Set<UInt64>> entry : producingValidatorsBySlot.entrySet()) {
+      final UInt64 slot = entry.getKey();
+      final Set<UInt64> producingValidators = entry.getValue();
+
+      producedSignatureCount +=
+          countProducedSignatures(
+              assignedSubcommitteeIndicesByValidatorIndex, slot, producingValidators);
+
+      final UInt64 inclusionSlot = slot.plus(1);
+      includedSignatureCountFutures.add(
+          getSyncAggregateAtSlot(inclusionSlot)
+              .thenApply(
+                  maybeSyncAggregate ->
+                      maybeSyncAggregate
+                          .map(
+                              syncAggregate ->
+                                  countIncludedSignatures(
+                                      assignedSubcommitteeIndicesByValidatorIndex,
+                                      slot,
+                                      producingValidators,
+                                      syncAggregate))
+                          .orElse(0)));
+    }
+
+    final int numberOfProducedSignatures = producedSignatureCount;
+    return SafeFuture.collectAll(includedSignatureCountFutures.stream())
+        .thenApply(
+            includedSignatureCounts -> includedSignatureCounts.stream().mapToInt(a -> a).sum())
+        .thenApply(
+            numberOfIncludedSignatures ->
+                new SyncCommitteePerformance(
+                    numberOfExpectedSignatures,
+                    numberOfProducedSignatures,
+                    numberOfIncludedSignatures));
+  }
+
+  private synchronized int countIncludedSignatures(
+      final Map<UInt64, Set<Integer>> assignedSubcommitteeIndicesByValidatorIndex,
+      final UInt64 slot,
+      final Set<UInt64> producingValidators,
+      final SyncAggregate syncAggregate) {
+
+    final SszBitvector syncCommitteeBits = syncAggregate.getSyncCommitteeBits();
+    int numberOfIncludedSignatures = 0;
+    for (UInt64 producingValidatorIndex : producingValidators) {
+      final Set<Integer> committeeIndices =
+          assignedSubcommitteeIndicesByValidatorIndex.get(producingValidatorIndex);
+      if (committeeIndices == null) {
+        LOG.debug(
+            "Validator {} produced a SyncCommitteeSignature in slot {} but wasn't expected to",
+            producingValidatorIndex,
+            slot);
+        continue;
+      }
+      for (Integer committeeIndex : committeeIndices) {
+        if (syncCommitteeBits.getBit(committeeIndex)) {
+          numberOfIncludedSignatures++;
```

Comment:
Is this just an estimate?  Seems like we could be double counting some of these?

Reply:
Above comment explains this best. This is counting the number of times you get rewarded. So it does double count if you're in the same subcommittee multiple times, but so do the rewards (and so does our expected calculation).
So for 64 validators in minimal spec I get:
```
Sync committee performance: expected 256, produced 256, included 256 (100%)
```
because I'm running all the validators in the sync committee - 32 validators in the sync committee and 8 slots per epoch.